src/compiler/commandLineParser.ts

```diff
@@ -3849,7 +3849,10 @@ function hasFileWithHigherPriorityExtension(file: string, literalFiles: Map<stri
         return false;
     }
     for (const ext of extensionGroup) {
-        if (fileExtensionIs(file, ext)) {
+        // d.ts files match with .ts extension and with case sensitive sorting the file order for same files with ts tsx and dts extension is
+        // d.ts, .ts, .tsx in that order so we need to handle tsx and dts of same same name case here and in remove files with same extensions
+        // So dont match .d.ts files with .ts extension
+        if (fileExtensionIs(file, ext) && (ext !== Extension.Ts || !fileExtensionIs(file, Extension.Dts))) {

```

Comment:
There's the same code in the function below this one; does it also need to change?

(I don't fully understand this change so I'll probably defer to someone else)

Reply:
No since if you look at ts order its `.ts, .tsx, .d.ts` so in this loop we first look at `.ts` In the below loop we iterate over in reverse order to remove file with the lower priority extensions so that part is will match `.d.ts` first and hence correct.